### PR TITLE
ACCSEC-22011 challenges order

### PR DESCRIPTION
**Pull request by yafuquen**
## Jira Ticket
- ACCSEC-22011

## Description
- Add order parameter to get challenge list

## Commit message
-  feat: Ordering for challenge list

## Screenshot
simulator_screenshot_F05BB628-F335-417B-B941-90278DDD49CC

## Testing
- [ ] Added unit tests
- [x] Ran unit tests successfully
- [x] Added documentation for public APIs and/or Wiki

### CODE REVIEW

File: TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ import Foundation
 protocol ChallengeProvider {
   func get(withSid sid: String, withFactor factor: Factor, success: @escaping ChallengeSuccessBlock, failure: @escaping FailureBlock)
   func update(_ challenge: Challenge, payload: String, success: @escaping ChallengeSuccessBlock, failure: @escaping FailureBlock)
-  func getAll(for factor: Factor, status: ChallengeStatus?, pageSize: Int, pageToken: String?, success: @escaping (ChallengeList) -> (), failure: @escaping FailureBlock)
+  func getAll(for factor: Factor, status: ChallengeStatus?, pageSize: Int, order: ChallengeListOrder, pageToken: String?, success: @escaping (ChallengeList) -> (), failure: @escaping FailureBlock)
```
Comment (AuthyApps) [inline TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift:25]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/9c5676df9e87745b9dd083abd2e86fcf45c18a45/TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift#L25"></span>Line should be 195 characters or less: currently 196 characters<br />
<code>line_length</code> <code>ChallengeRepository.swift:25</code></td>
    </tr>
  </tbody>
</table>

File: TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift [hunk 2]
```diff
@@ -87,8 +87,8 @@ extension ChallengeRepository: ChallengeProvider {
     }, failure: failure)
   }
   
-  func getAll(for factor: Factor, status: ChallengeStatus?, pageSize: Int, pageToken: String?, success: @escaping (ChallengeList) -> (), failure: @escaping FailureBlock) {
-    apiClient.getAll(forFactor: factor, status: status?.rawValue, pageSize: pageSize, pageToken: pageToken, success: { [weak self] response in
+  func getAll(for factor: Factor, status: ChallengeStatus?, pageSize: Int, order: ChallengeListOrder, pageToken: String?, success: @escaping (ChallengeList) -> (), failure: @escaping FailureBlock) {
```
Comment (AuthyApps) [inline TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift:90]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/9c5676df9e87745b9dd083abd2e86fcf45c18a45/TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift#L90"></span>Line should be 195 characters or less: currently 198 characters<br />
<code>line_length</code> <code>ChallengeRepository.swift:90</code></td>
    </tr>
  </tbody>
</table>

File: TwilioVerifySDKTests/TwilioVerify/Sources/Domain/Challenge/Mocks/ChallengeProviderMock.swift [hunk 3]
```diff
@@ -47,7 +47,7 @@ extension ChallengeProviderMock: ChallengeProvider {
     }
   }
   
-  func getAll(for factor: Factor, status: ChallengeStatus?, pageSize: Int, pageToken: String?, success: @escaping (ChallengeList) -> (), failure: @escaping FailureBlock) {
+  func getAll(for factor: Factor, status: ChallengeStatus?, pageSize: Int, order: ChallengeListOrder, pageToken: String?, success: @escaping (ChallengeList) -> (), failure: @escaping FailureBlock) {
```
Comment (AuthyApps) [inline TwilioVerifySDKTests/TwilioVerify/Sources/Domain/Challenge/Mocks/ChallengeProviderMock.swift:50]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/9c5676df9e87745b9dd083abd2e86fcf45c18a45/TwilioVerifySDKTests/TwilioVerify/Sources/Domain/Challenge/Mocks/ChallengeProviderMock.swift#L50"></span>Line should be 195 characters or less: currently 198 characters<br />
<code>line_length</code> <code>ChallengeProviderMock.swift:50</code></td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.18%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.62%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerifySDK/TwilioVerify/Sources/Models/ChallengeListPayload.swift | **`100.00%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.14%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioVerify/Sources/API/ChallengeAPIClient.swift | **`93.62%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`93.65%`**
TwilioVerifySDK/TwilioVerify/Sources/Models/ChallengeListPayload.swift | **`100.00%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by AlejandroOrozco — APPROVED** [review]
LG 👍

**Review by SergioFierro — APPROVED** [review]

**Pull request merged**